custom_components/rrd/camera.py

```diff
@@ -72,20 +72,42 @@ def __init__(self, name, rrd, width, height, timerange, args):
         self._defs = []
         self._lines = []
         try:
-            info = rrdtool.info(self._rrd)
-            self._step = info["step"]
-            for key, value in info.items():
+            rrdinfo = rrdtool.info(self._rrd)
+            self._step = rrdinfo["step"]
+            for key, value in rrdinfo.items():
                 if ".index" in key:
-                    ds = re.search(r"\[(.*?)\]", key).group(1)
+                    ds = re.search(r"\[(.*?)\]", key).group(1)  # Datasource name
                     self._unique_id += f"_{ds}"
-                    cf = info[f"rra[{value}].cf"]
-                    self._defs.append(f"DEF:{ds.capitalize()}={self._rrd}:{ds}:{cf}")
+                    cf = rrdinfo[f"rra[0].cf"]
```

Comment:
why 0  and not value ?

Reply:
Each DS has index (0,1,2,...). The `value` variable contains this DS `index`. This index defines order of DS (when multiple DS is in a file). It is not pointing to primary RRA. RRA definition is 100% same for all DS.

There is technicaly nothing like primary RRA, but the reality is the FIRST (`rra[0]`) RRA contains not agregated data (or the least agregated data). It is the reason, why `[0]` is used there.

The previous code `cf = info[f"rra[{value}].cf"]` did not make sense.. It could cause a bug: It could fail, when there is more DS than RRA.

Adding `rrdinfo` example of file with 3 DS. Every DS has 2 RRAs:
```
filename = "filename.rrd"
rrd_version = "0003"
step = 300
last_update = 1603567483
header_size = 1536
ds[value1].index = 0
ds[value1].type = "GAUGE"
ds[value1].minimal_heartbeat = 600
ds[value1].min = NaN
ds[value1].max = NaN
ds[value1].last_ds = "U"
ds[value1].value = NaN
ds[value1].unknown_sec = 283
ds[value2].index = 1
ds[value2].type = "GAUGE"
ds[value2].minimal_heartbeat = 600
ds[value2].min = NaN
ds[value2].max = NaN
ds[value2].last_ds = "U"
ds[value2].value = NaN
ds[value2].unknown_sec = 283
ds[value3].index = 2
ds[value3].type = "GAUGE"
ds[value3].minimal_heartbeat = 600
ds[value3].min = NaN
ds[value3].max = NaN
ds[value3].last_ds = "U"
ds[value3].value = NaN
ds[value3].unknown_sec = 283
rra[0].cf = "AVERAGE"
rra[0].rows = 24
rra[0].cur_row = 12
rra[0].pdp_per_row = 12
rra[0].xff = 5.0000000000e-01
rra[1].cf = "AVERAGE"
rra[1].rows = 365
rra[1].cur_row = 217
rra[1].pdp_per_row = 288
rra[1].xff = 5.0000000000e-01
```

Does it make sense?

Reply:
```
for key, value in rrdinfo.items():
  if ".index" in key:
```
This `for` command is there for creating `DEF:` for datasources `value1`, `value2`, `value3` (in my example).

In my example this code will generate these `DEF`s:

```
DEF:Value1=rrdfile.rrd:value1:AVERAGE                           # Primary DEF (same as in released `rrd_recorder` version)
DEF:Value1_AVERAGE_288=rrdfile.rrd:value1:AVERAGE:step=86400    # Secondary DEF based on secondary RRA (New feature)

DEF:Value2=rrdfile.rrd:value2:AVERAGE                           # Primary DEF (same as in released `rrd_recorder` version)
DEF:Value2_AVERAGE_288=rrdfile.rrd:value2:AVERAGE:step=86400    # Secondary DEF based on secondary RRA (New feature)

DEF:Value3=rrdfile.rrd:value2:AVERAGE                           # Primary DEF (same as in released `rrd_recorder` version)
DEF:Value3_AVERAGE_288=rrdfile.rrd:value2:AVERAGE:step=86400    # Secondary DEF based on secondary RRA (New feature)
```

Not sure, if my explanation of `for` purpose is clean. Just ask me, if not clean. :)

Reply:
My point was that `value` is no longer used, no point in using `items()` to extract it, we only need to iterate the `key`